pgxn/remotexact/remotexact.c

```diff
@@ -157,25 +164,34 @@ rwset_add_region(int region)
 }
 
 static void
-rx_collect_relation(int region, Oid dbid, Oid relid)
+rx_collect_relation(int region, Oid dbid, Oid relid, char relkind)
 {
 	CollectedRelation *relation;
+	RemoteRelkind rrelkind = get_remote_relkind(relkind);
+
+	Assert(rrelkind != REMOTE_RELKIND_INVALID);
 
 	init_rwset_collection_buffer(dbid);
 
 	relation = get_collected_relation(relid, true);
 	relation->region = region;
-	relation->is_index = false;
-	relation->is_table_scan = true;
+	relation->is_index = rrelkind == REMOTE_RELKIND_INDEX;
```

Comment:
I'm considering storing the `rrelkind` directly in the rwset and let the validation code handle the rest next time I revisit this part.

Reply:
Makes sense.